locations/storefinders/go_review.py

```diff
@@ -0,0 +1,69 @@
+import re
+
+from scrapy.linkextractors import LinkExtractor
+from scrapy.spiders import CrawlSpider, Rule
+
+from locations.hours import OpeningHours
+from locations.items import Feature
+
+
+class GoReviewSpider(CrawlSpider):
+    """
+    To use this spider, specify one or more start_urls,
+    normally something like "https://hlinfo.goreview.co.za/store-locator/store-information"
+
+    Also known as socialplaces.io or Social Places
+    """
+
+    custom_settings = {"ROBOTSTXT_OBEY": False}  # robots.txt disallows everything
+    rules = [
+        Rule(
+            LinkExtractor(allow=r"^https:\/\/.*?\d+\.goreview\.co\.za\/store-information\?store-locator=.*$"),
+            callback="parse",
+        )
+    ]
+
+    def parse(self, response):
+        properties = {
+            "ref": re.sub(r"\.goreview\.co\.za.*", "", re.sub(r"https:\/\/", "", response.url)),
+            "branch": response.xpath('//div[@class="left-align-header"]/h2/text()')
+            .get()
+            .replace(self.item_attributes["brand"], "")
+            .strip(),
+            "addr_full": re.sub(
+                r"\s+",
+                " ",
+                ", ".join(
+                    filter(
+                        None,
+                        map(
+                            str.strip,
+                            response.xpath(
+                                '//div[contains(@class, "content-wrapper")]/div[2]/div[1]//p/text()'
+                            ).getall(),
+                        ),
+                    )
+                ),
+            ),
+            "phone": response.xpath(
+                '//div[contains(@class, "content-wrapper")]/div[2]/div[3]//a[contains(@href, "tel:")]/@href'
+            ).get(),
+            "facebook": response.xpath(
+                '//div[contains(@class, "content-wrapper")]/div[2]/div[5]//a[contains(@href, "facebook.com")]/@href'
+            ).get(),
+            "website": response.url,
+        }
+        if maps_links_js := response.xpath('//script[contains(text(), "#apple_maps_directions")]/text()').get():
+            if "&sll=" in maps_links_js:
+                properties["lat"], properties["lon"] = maps_links_js.split("&sll=", 1)[1].split("&", 1)[0].split(",", 2)
+        hours_string = " ".join(
+            response.xpath('//div[contains(@class, "content-wrapper")]/div[2]/div[4]/p//text()').getall()
+        )
+        properties["opening_hours"] = OpeningHours()
+        properties["opening_hours"].add_ranges_from_string(hours_string)
```

Comment:
Is it possible for the opening hours to be specified in a language other than English in the future? Should there be a storefinder attribute `days = DAYS_EN` and then use `properties["opening_hours"].add_ranges_from_string(hours_string, days=self.days)` here to allow possible override of opening hours language?

Reply:
Could be possible, done